server/src/main/java/org/elasticsearch/action/support/PlainListenableActionFuture.java

```diff
@@ -43,8 +43,9 @@ protected void done(boolean success) {
         synchronized (this) {
             executedListeners = true;
         }
-        Object listeners = this.listeners;
+        final Object listeners = this.listeners;
         if (listeners != null) {
+            this.listeners = null;
```

Comment:
Not completely related, but seems useful to discard the completed listeners here now that we no longer care about them. At least if we kept the listener around long after it were completed then this would avoid pointlessly holding those listeners.

Reply:
Just a drive-by comment, but this variable shadowing makes my eyes twitch. Any chance we can use a different than than `listeners` for the local var?

Reply:
Dammit @dakrone I started to tidy that up and then found another thing to clean, and then another, and now I have a whole separate PR: #68735.